tasks/banana.js

```diff
@@ -13,19 +13,40 @@ module.exports = function ( grunt ) {
 			messageCount = 0,
 			ok = true;
 
+		if ( this.filesSrc.length === 0 ) {
+			grunt.log.error( 'Target directory does not exist.' );
+			ok = false;
+			return;
```

Comment:
Unlike the below code, which is inside a forEach loop and returning from there. This code is outside that loop and directly in the grunt task function. As such, returning from here short-circuits the task and doesn't result in `ok` being evaluated to return false. Either omit the return statement and let the below code run (so that it eventually returns false), or change this to `return false;` (and no need to set `ok`).

I'd recommend the latter, and maybe also move `ok = true;` to right before the forEach-loop starts.